3rdparty/CMakeLists.txt

```
@@ -525,6 +530,36 @@ if(WITH_LIBTBB STREQUAL "EXTERNAL")
   endif(UNIX)
 endif()
 
+# libFaiss
+set(LIBFAISS_TARGET libfaiss)
+set(LIBFAISS_INCLUDES ${3RDPARTY_LIBS_DIR}/Source/libfaiss/faiss)
+set(LIBFAISS_LIBS ${3RDPARTY_LIBS_DIR}/Build/libfaiss/faiss/libfaiss.a)
+ADD_BY_PRODUCTS_VARIABLE ("LIBFAISS" ${LIBFAISS_LIBS})
+if(WITH_FAISS STREQUAL "EXTERNAL")
+  externalproject_add(
+        ${LIBFAISS_TARGET}
+        URL ${WITH_FAISS_URL}
+        LOG_CONFIGURE TRUE
+        LOG_BUILD TRUE
+        LOG_INSTALL TRUE
+        TIMEOUT 600
+        DOWNLOAD_NO_PROGRESS 1
+        BUILD_BYPRODUCTS ${3RDPARTY_LIBS_DIR}/Build/libfaiss/faiss/libfaiss.a
+        CONFIGURE_COMMAND ${CMAKE_COMMAND} -B <BINARY_DIR> -S <SOURCE_DIR>
+        "-DCMAKE_CXX_FLAGS=-llapack -lopenblas"
+        -DCMAKE_INSTALL_PREFIX=${3RDPARTY_LIBS_DIR}/faiss
+        -DBUILD_SHARED_LIBS=OFF
+        -DFAISS_ENABLE_GPU=OFF
+        -DFAISS_ENABLE_PYTHON=OFF
+        -DBUILD_TESTING=OFF
+        BUILD_COMMAND make -C <BINARY_DIR> -j faiss
+  )
```

```suggestion
        BUILD_COMMAND make -C <BINARY_DIR> -j faiss
        "${LIBFAISS_BYPRODUCTS}"
  )
```
다음은 상단의 ADD_BY_PRODUCTS_VARIABLE 매크로의 용도입니다.

ADD_BY_PRODUCTS_VARIABLE에 첫번째 인수 (XXX) 에 대해서, Ninja 빌드이면 "${XXX_BYPRODUCTS}" 에 대해서  BUILD_BYPRODUCTS를 설정합니다. (MAKE 호환성을 위함)